File: 1stPhase/Armstrong.c
#include <stdio.h>
int main()
{
    int n = 153, r, sum = 0, dummy;
    dummy = n;
    while (n > 0)
    {
        r = n % 10;
        sum = sum + (r * r * r);
        n = n / 10;
    }
    if (dummy == sum)
    {
        printf("Armstrong number");
    }
    else
    {
        printf(" Not Armstrong number");
    }
    return 0;
}

File: JavaPrograms/Armstrong.java
public class Armstrong {
    public static void main(String[] args) {
        int number = 153; 
        if (isArmstrong(number)) {
            System.out.println(number + " is an Armstrong number.");
        } else {
            System.out.println(number + " is not an Armstrong number.");
        }
    }

    public static boolean isArmstrong(int number) {
        int originalNumber, remainder, result = 0, n = 0;

        originalNumber = number;
        for (;originalNumber != 0; originalNumber /= 10, ++n) {}

        originalNumber = number;
        for (;originalNumber != 0; originalNumber /= 10) {
            remainder = originalNumber % 10;
            result += Math.pow(remainder, n);
        }

        return result == number;
    }
}
